resolver-dns/src/test/java/io/netty/resolver/dns/SearchDomainTest.java

```diff
@@ -33,20 +33,19 @@
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringContains.containsString;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
-import static org.hamcrest.Matchers.not;
-import static org.hamcrest.core.StringContains.containsString;
 
 public class SearchDomainTest {
 
     private DnsNameResolverBuilder newResolver() {
         return new DnsNameResolverBuilder(group.next())
             .channelType(NioDatagramChannel.class)
-            .nameServerAddresses(DnsServerAddresses.singleton(dnsServer.localAddress()))
```

Comment:
My preference is to break the interface, and force the user to deal with the change in behavior as opposed to silently change behavior behind the scenes. `@deprecated` methods are easy to miss and may result in unexpected/undesirable behavior (as indicated in the issue which motivated this PR). `@deprecated` methods also create additional maintain burden and may limit design decisions. IMO since the API is unstable we should take advantage and make the change.